src/zebras/_split_frame.py

```diff
@@ -0,0 +1,283 @@
+# Copyright 2018 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License").
+# You may not use this file except in compliance with the License.
+# A copy of the License is located at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# or in the "license" file accompanying this file. This file is distributed
+# on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+# express or implied. See the License for the specific language governing
+# permissions and limitations under the License.
+
+from __future__ import annotations
+
+import dataclasses
+from typing import Optional
+
+import numpy as np
+from toolz import first, keymap, valmap, dissoc, merge
+
+from gluonts import maybe
+
+from ._base import Pad
+from ._period import Periods, period
+from ._repr import html_table
+from ._util import _replace
+
+
+@dataclasses.dataclass
+class SplitFrame:
+    _past: dict
+    _future: dict
+    index: Optional[Periods]
+    static: dict
+    past_length: int
+    future_length: int
+    tdims: dict
+    metadata: Optional[dict] = None
+    default_tdim: int = -1
+    _pad: Pad = Pad()
+
+    @property
+    def past(self):
+        return TimeFrame(
+            self._past,
+            index=maybe.map(
+                self.index, lambda index: index[: self.past_length]
+            ),
+            static=self.static,
+            length=self.past_length,
+            tdims=self.tdims,
+            metadata=self.metadata,
+            _pad=Pad(self._pad.left, 0),
+        )
+
+    @property
+    def future(self):
+        return TimeFrame(
+            self._future,
+            index=maybe.map(
+                self.index, lambda index: index[-self.future_length :]
+            ),
+            static=self.static,
+            length=self.future_length,
+            tdims=self.tdims,
+            metadata=self.metadata,
+            _pad=Pad(0, self._pad.right),
+        )
+
+    def __len__(self):
+        return self.past_length + self.future_length
+
+    def set(self, name, value, tdim=None):
+        tdim = maybe.unwrap_or(tdim, self.default_tdim)
+        assert value.shape[tdim] == len(self)
+
+        past, future = np.split(
+            value,
+            [self.past_length],
+            axis=tdim,
+        )
+
+        return _replace(
+            past=merge(self.past, {name: past}),
+            future=merge(self.future, {name: future}),
+            tdims=merge(self.tdims, {name: tdim}),
+        )
+
+    def set_like(self, ref: str, column, value, tdim=None):
```

Comment:
The idea is to be able to do `set` on both `SplitFrame` and `TimeFrame`. Say I want to add an observed values indicator for a column. In case of `SplitFrame`, we need to know whether that value is `past`, `future` or both. So for `TimeFrame` this method is just like `set`, but for `SplitFrame` that gets translated into either `set_past`, `set_future` or `set`.